### PR TITLE
docs: content enhancements regarding client and server

### LINKED ISSUE #34: docs: reana-cluster
Update "Getting started" docs following up the release of `reana-cluster`. https://github.com/reanahub/reana-cluster/issues/3

### LINKED ISSUE #35: docs: reana-client
Update "Getting started" docs following up the release of [reana-client](https://github.com/reanahub/reana-client).

### PR DESCRIPTION
**Pull request by tiborsimko**
* Amends documentation structure, introduces several new sections such as
  Concepts, User guide, Administrator guide, and Developer guide.

* Removes obsoleted ``reana`` installation script in profit of
  ``reana-cluster``. (closes #34)

* Advertises ``reana-client`` for end users. (closes #35)

Signed-off-by: Tibor Simko <tibor.simko@cern.ch>

### CODE REVIEW

File: docs/concepts.rst [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+Concepts
+========
+
+Reusable analyses
+-----------------
+
+The revalidation, reinterpretation and reuse of research data analyses requires
+having access not only to the original experimental datasets and the analysis
+software, bit also to the operating system environment and the computational
```
Comment (diegodelemos) [inline docs/concepts.rst]: ".. but also to the operating .."

File: docs/concepts.rst [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+It is advised to structure your analysis code using the "inputs", "code",
+"outputs", "environments", "workflows" strategy. For example:
+
+.. code-block:: console
+
+   $ ls -l .
+   inputs/mydata.csv
+   code/mycode.py
+   environments/myenvironment/Dockerfile
```
Comment (diegodelemos) [inline docs/concepts.rst]: I don't know if it is only me, but this "example" sounds kind of ok in general, but `myenvironment` specifically sounds way too abstract

File: docs/concepts.rst [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    workflow:
+      type: cwl
+      file: workflow/myworkflow.cwl
+
+REANA platform
+--------------
+
+Users can use ``reana-client`` command line client that talks to
+``reana-server`` REST API sending commands to REANA cloud. The REANA cloud can
+be deployed using ``reana-cluster`` command line client. THe REANA cluster
```
Comment (diegodelemos) [inline docs/concepts.rst]: The instead of THe

File: docs/concepts.rst [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+consists of several micro-service components that manage and run user-supplied
+computational workflows. Together, the (local) REANA client and (remote) REANA
+cluster constitute the REANA reusable analysis platform.
```
Comment (diegodelemos) [inline docs/concepts.rst]: here we use REANA cluster, in some other places we use REANA cloud, what's the status with this terminology?

File: docs/userguide.rst [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+- ``REANA_SERVER_URL`` Permits to specify to which REANA cloud instance the
+  client should connect. For example:
+
+.. code-block:: console
+
+   $ export REANA_SERVER_URL=http://reana.cern.ch
+
+- ``REANA_WORKON`` Permits to specify a concrete workflow ID run for the given
+  analysis. (As an alternative to specifying ``--workflow-id`` in commands.) For
```
Comment (diegodelemos) [inline docs/userguide.rst]: currently it is called `--workflow` see [here](https://github.com/reanahub/reana-client/blob/master/reana_client/cli/workflow.py#L176) should we open an issue for renaming or we change it here?

File: docs/concepts.rst (hunk 4 above)
Comment (tiborsimko) [reply docs/concepts.rst]: Briefly, "cloud" in user speak context, "cluster" in IT-admin speak context.

File: docs/userguide.rst (hunk 5 above)
Comment (tiborsimko) [reply docs/userguide.rst]: Amending to ``-workflow`` here and in ``reana-client`` as wel.

File: docs/concepts.rst (hunk 2 above)
Comment (tiborsimko) [reply docs/concepts.rst]: Yeah, it looks a bit cryptic... What about using `mypython` or `mybuntu` or something more understandable like that?

**Pull request merged**